### ISSUE TITLE
Various tables advertise an index, but do not implement one

### ISSUE THREAD

**Issue by offlinemark** [osquery/osquery#5675]
If a table includes `index=True` in a column in its specfile, the expectation is that the table's generation function will implement a special case that is optimized for retrieving rows given a constraint on the column. This becomes relevant when tables are using in joins where the sqlite query planner may opt to repeatedly query a table, constraining a certain indexed column, expecting that to be efficient. 

There are many tables that advertise an index (use `index=True`), however, do not implement the special case. We should probably either remove or implement the index for these tables.

Here are a few of the tables and the offending column:

```
chrome_extensions identifier
browser_plugins identifier
crashes parent
event_taps event_tap_id
kernel_extensions idx
package_receipts package_id
package_receipts version
preferences subkey
safari_extensions identifier
ec2_instance_metadata instance_id
authorized_keys key_file
firefox_addons identifier
known_hosts key_file
opera_extensions identifier
ssh_configs option
ssh_configs ssh_config_file
smart_drive_info device_name
drivers device_id
ie_extensions registry_path
logon_sessions logon_id
winbaseobj session_id
winbaseobj object_name
yara matches
```

This may not be a complete list. It was generated with this imprecise, grep-based python script: https://gist.github.com/mossberg/37334234b5c43fe58dbbdbe87a7a7f44

To start, we just need to go through these, confirm whether the table uses this particular column (e.g. via `context.constraints["identifier"].exists(EQUALS)`), and if does not, delete the `index=True` from the specfile. In the future, we can add it back when the index is implemented.

In the future, it might be nice to integrate this into CI or tests somehow. The tricky part is being able to conclusively check whether a table implements the index or not.

**Comment by directionless** [comment]
I suspect there's some overlap here with https://github.com/osquery/osquery/issues/5668

**Comment by mike-myers-tob** [comment]
Ok so the problematic tables were fixed by @packetzero in PR #5938 but this issue is still open because osquery needs a CI test that checks if a table is actually implementing an index that it advertises.

**Comment by directionless** [comment]
> If a table includes index=True in a column in its specfile, the expectation is that the table's generation function will implement a special case that is optimized for retrieving rows given a constraint on the column.

I'm not sure we've wholly settled on that premise. @theopolis and I have been discussing it.

**Comment by theopolis** [comment]
> Ok so the problematic tables were fixed by @packetzero in PR #5938

Some of them were fixed, but not all and this created a significant bug in 4.1.1 and thus the premise for 4.1.2.

We needed to make a change to how the column options are enforced and linted in #6104. And then update the remaining tables in #6105, #6106, and #6122.

A helpful next-step is going through all of the open integration test issues and closing out the ones that now have an integration test. We have been adding more and more without following up on the issues (bad house cleaning). Then finally we should finish up with the integration testing and make sure all tables have a test. We can then close this out and have decent confidence we will not re-introduce.
